osf/management/commands/reindex_provider.py

```diff
@@ -1,24 +1,30 @@
-"""Resend all preprints for given providers to SHARE."""
+"""Resend all resources (nodes, registrations, preprints) for given providers to SHARE."""
 import logging
 
 from django.core.management.base import BaseCommand
-from osf.models import PreprintProvider
+from osf.models import AbstractProvider, AbstractNode, Preprint
 from api.share.utils import update_share
 
 logger = logging.getLogger(__name__)
 
+
 def reindex_provider(provider):
     logger.info('Sending {} preprints to SHARE...'.format(provider.preprints.count()))
-    for preprint in provider.preprints.all():
+    for preprint in Preprint.objects.filter(provider=provider):
         update_share(preprint)
 
+    logger.info('Sending {} AbstractNodes to SHARE...'.format(AbstractNode.objects.filter(provider=provider).count()))
+    for abstract_node in AbstractNode.objects.filter(provider=provider):
+        update_share(abstract_node)
+
+
 class Command(BaseCommand):
     def add_arguments(self, parser):
         super(Command, self).add_arguments(parser)
         parser.add_argument('providers', type=str, nargs='+', help='Provider _ids')
 
     def handle(self, *args, **options):
         provider_ids = options.get('providers', [])
-        for provider in PreprintProvider.objects.filter(_id__in=provider_ids):
+        for provider in AbstractProvider.objects.filter(_id__in=provider_ids):
```

Comment:
Drive by: the only downside to doing it this way is that, in the case of reindexing osf registries, we're going to end up having to re-index all of osf preprints as well.

Reply:
Good point, this may actually be a good time to change the providers `_id` because they should really be unique, though this would take some effort.

Reply:
Perhaps an arg to specify which provider type you want to target and filter on type `AbstractProvider.objects.filter(type__contains='preprint', _id__in=provider_ids)`